### PR TITLE
AccessControl: Protect org users lookup

### PR DESCRIPTION
**Pull request by gamab**
**What this PR does / why we need it**:
This issue is needed to cover `/api/org/users/lookup` endpoint with accesscontrol permissions.
In the meantime, I moved the legacy accesscontrol check to the middleware layer.
This PR will help in fixing https://github.com/grafana/grafana-enterprise/issues/1862.

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

### CODE REVIEW

**Review by alexanderzobnin — APPROVED** [review]
Looks good!

**Comment by vtorosyan** [comment]
Is this supposed to go with 8.2?

**Review by IevaVasiljeva — COMMENTED** [review]
I tested this and it all seems to work fine, apart from team/folder admins not having the lookup permissions anymore when the new access control is enabled. But I guess that is intentional, as customers are expected to create their own FGAC roles for team/folder administration?

File: pkg/api/org_users_test.go [hunk 1]
```diff
@@ -176,3 +128,123 @@ func TestOrgUsersAPIEndpoint_userLoggedIn(t *testing.T) {
...
+	}
+	sc.m.Use(pretendSignInMiddleware)
+
+	hs.registerRoutes()
+	hs.RouteRegister.Register(sc.m.Router)
+
+	return sc, db
+}
+
+func TestOrgUsersAPIEndpoint_LegacyAccessControl_TeamAdmin(t *testing.T) {
```
[the remark this answers is not in the document]
Comment (gamab) [reply pkg/api/org_users_test.go:162]: Yes, very good point. TBO the test wasn't there before and I did not take time to evaluate the setup difficulty of this test :slightly_smiling_face:

**Comment by gamab** [comment]
> Is this supposed to go with 8.2?

Yes it is, to make sure the permissions tab for data sources works as expected :)

**Comment by gamab** [comment]
> I tested this and it all seems to work fine, apart from team/folder admins not having the lookup permissions anymore when the new access control is enabled. But I guess that is intentional, as customers are expected to create their own FGAC roles for team/folder administration?

You raise a good point there. I believe assigning permissions to team/folder admins will come later. But now, I wonder if the scope of the permission I protect the endpoint with `{"action": "org.users:read", "scope": "users:*"}` is not too wide for them... :thinking:
I guess it's ok to leave as is and to change it later, when we handle the team/folder admins case. It's better to have this over protected than under protected.
Another thing I need to do though, is to add the permission to  `fixed:datasources:permissions:admin`.

File: pkg/api/org_users_test.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (gamab) [reply pkg/api/org_users_test.go:205]: Done.

**Pull request merged**